### PR TITLE
[GEP-18] Revendor `gardener/gardener@v1.45`, cleanup token projection artefacts, adapt to new secrets manager

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area security
/kind enhancement
/platform azure

**What this PR does / why we need it**:

- Vendors g/g and adapts to all breaking changes introduced by https://github.com/gardener/gardener/pull/5803
- Cleans up a lot of migration code around moving to short-lived serviceaccount tokens introduced with #421
- Adapts this extension to the new secrets manager, which makes it ready for supporting the shoot CA rotation feature

**Which issue(s) this PR fixes**:
Part of https://github.com/gardener/gardener/issues/3292

**Special notes for your reviewer**:

Use the dedicated commits, otherwise this PR is pretty much unreviewable.

Based on https://github.com/gardener/gardener-extension-provider-aws/pull/538 (thanks @timebertt for structuring it so well).

**Release note**:

```breaking operator
This extension is only compatible with Gardener versions `>= v1.37`.
```
```feature operator
This extension is prepared to support the Shoot CA rotation feature ([GEP-18](https://github.com/gardener/gardener/issues/3292)).
```

### CODE REVIEW

**Review by kris94 — COMMENTED** [review]
/lgtm

File: charts/gardener-extension-provider-azure/templates/deployment.yaml [hunk 1]
```diff
@@ -55,9 +55,7 @@ spec:
         - --disable-controllers={{ .Values.disableControllers | join "," }}
         - --disable-webhooks={{ .Values.disableWebhooks | join "," }}
         - --health-bind-address=:{{ .Values.healthPort }}
-        {{- if .Values.gardener.version }}
```
Comment (timebertt) [inline charts/gardener-extension-provider-azure/templates/deployment.yaml:58]: I forgot to clean this up in https://github.com/gardener/gardener-extension-provider-aws/pull/538.
I will perform this for provider-aws as well.

**Review by timebertt — DISMISSED** [review]
Thank you!
/lgtm

File: charts/gardener-extension-provider-azure/templates/deployment.yaml (hunk 1 above)
Comment (timebertt) [reply charts/gardener-extension-provider-azure/templates/deployment.yaml:58]: Done in https://github.com/gardener/gardener-extension-provider-aws/pull/542

**Review by timebertt — APPROVED** [review]
/lgtm

**Pull request merged**